### PR TITLE
OEUI-225: Edit active drug orders from the orders home page

### PR DESCRIPTION
**Pull request by rhenshaw56**
### **JIRA TICKET NAME**
[OEUI-225](https://issues.openmrs.org/browse/OEUI-225) Edit active drug orders from the orders home page

### **Summary**
- implemented edit feature for active drug order

## I have checked that on this Pull request

- [x] I can successfully create Drug orders
- [x] I can sucessfully create Lab orders
- [x] I can sucessfully search for drug orders

### CODE REVIEW

File: tests/components/orderentry/OrdersTable.test.jsx [hunk 1]
```diff
@@ -73,4 +73,50 @@ describe('Orders component test-suite', () => {
     const wrapper = mount(<ConnectedOrdersTablePage store={store} {...props}/>);
     expect(wrapper.length).toBe(1);
   });
+  it('handles edting of active orders', () => {
```
Comment (jomadoye) [inline tests/components/orderentry/OrdersTable.test.jsx]: editing :-)

File: tests/reducers/orderSelectionReducer.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            selectedOrder: {},
+        },
+    };
+    const action = {
+    type: 'SOME_OTHER_ACTION_TYPE',
+    };
+
+    const newState = orderSelectionReducer(undefined, action);
+    expect(newState).toEqual(initialState.orderSelection);
+});
```
Comment (jomadoye) [inline tests/reducers/orderSelectionReducer.test.js]: Small styling issue here

**Comment by IAMOTZ** [comment]
Hi @rhenshaw56 
Please rebase off the master branch so that the recent features can be tested if they work well with this implementation of yours.

File: app/js/components/drugOrderEntry/ActiveOrders.jsx [hunk 3]
```diff
@@ -298,13 +308,17 @@ ActiveOrders.propTypes = {
...
+  activity: PropTypes.string,
 };
 
 ActiveOrders.defaultProps = {
   drugOrder: {
     loading: false,
     activeOrders: [],
   },
+  selectedOrder: {},
+  activity: '',
```
Comment (IAMOTZ) [inline app/js/components/drugOrderEntry/ActiveOrders.jsx]: I don't think the changes in the file are needed since this component(`ActiveOrders`) would be removed as the Active Orders are already displayed on the Orders List page.
If it won't break anything, I suggest you revert the changes.

File: tests/components/orderentry/OrdersTable.test.jsx (hunk 1 above)
Comment (rhenshaw56) [reply tests/components/orderentry/OrdersTable.test.jsx]: done

File: tests/reducers/orderSelectionReducer.test.js (hunk 2 above)
Comment (rhenshaw56) [reply tests/reducers/orderSelectionReducer.test.js]: done

File: app/js/actions/orderAction.js [hunk 4]
```diff
@@ -1,11 +1,16 @@
-import { SET_ORDER_ACTION } from './actionTypes';
+import * as types from './actionTypes';
```
Comment (dkayiwa) [inline app/js/actions/orderAction.js]: We generally do not recommend using wildcard imports https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotusewildcardimports

File: app/js/components/drugOrderEntry/index.jsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ import DraftDataTable from './addForm/DraftDataTable';
 import { selectDrugSuccess } from '../../actions/drug';
 import './styles.scss';
 
-export class SearchAndAddOrder extends React.Component {
+export class SearchAndAddOrder extends React.PureComponent {
```
Comment (dkayiwa) [inline app/js/components/drugOrderEntry/index.jsx]: Thanks for changing this to a PureComponent. 👍

File: app/js/components/orderEntry/OrdersTable.jsx [hunk 6]
```diff
@@ -7,6 +7,8 @@ import OrderHeader from './OrderHeader';
 import LabOrderDetails from './LabOrderDetails';
 import DrugOrderDetails from './DrugOrderDetails';
 import fetchOrders from '../../actions/fetchOrders';
+import * as orderTypes from './orderTypes';
```
Comment (dkayiwa) [inline app/js/components/orderEntry/OrdersTable.jsx]: Same here for wildcard imports.

File: app/js/reducers/orderSelectionReducer.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import * as types from '../actions/actionTypes';
```
Comment (dkayiwa) [inline app/js/reducers/orderSelectionReducer.js]: Same here

File: app/js/actions/orderAction.js (hunk 4 above)
Comment (rhenshaw56) [reply app/js/actions/orderAction.js]: noted! will revise

File: app/js/components/orderEntry/OrdersTable.jsx (hunk 6 above)
Comment (rhenshaw56) [reply app/js/components/orderEntry/OrdersTable.jsx]: fixed!

File: app/js/reducers/orderSelectionReducer.js (hunk 7 above)
Comment (rhenshaw56) [reply app/js/reducers/orderSelectionReducer.js]: fixed!

File: app/js/actions/orderAction.js (hunk 4 above)
Comment (rhenshaw56) [reply app/js/actions/orderAction.js]: fixed!

File: app/js/components/drugOrderEntry/index.jsx [hunk 8]
```diff
@@ -24,6 +24,20 @@ export class SearchAndAddOrder extends React.Component {
     draftOrder: {},
   };
 
+  componentWillMount() {
```
Comment (Efosaok) [inline app/js/components/drugOrderEntry/index.jsx]: The `componentWillMount` lifeCycle method is marked as `legacy`, therefore, it is advised not to be used in new code.please use `componentDidMount` instead.

File: tests/components/tabs/Tab.test.js [hunk 9]
```diff
@@ -8,13 +8,27 @@ describe('`Tab` component', () => {
...
+        const props = {
+            isActive: true,
+            handleOnClick: jest.fn(),
+            tabName: 'random tab name',
+            tabIndex: 2,
+        }
+        const wrapper = shallow(<Tab {...props} />);
+        const tab = wrapper.find('a');
+        tab.simulate('click', event);
+        expect(props.handleOnClick.mock.calls.length).toBe(0);
```
Comment (Efosaok) [inline tests/components/tabs/Tab.test.js]: is this assertion not supposed to be

`expect(props.handleOnClick.mock.calls.length).toBe(1);` ??

the test-suite's message says that the `isActive` prop is set to true when the click event is triggered, but here you are setting the isActive prop key to true when the button hasn't been clicked. there is no assertion for the state change as well.

File: app/js/reducers/orderSelectionReducer.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+const orderSelectionReducer = (state = initialState.orderSelection, action) => {
+  switch (action.type) {
+    case SET_SELECTED_ORDER:
+      return {
+        ...state,
+        currentOrderType: action.currentOrderType,
+        selectedOrder: action.order,
+        activity: action.activity,
+      };
```
Comment (Efosaok) [inline app/js/reducers/orderSelectionReducer.js]: why don't you just `destructure` instead of repeating `action` in these three instances..

File: tests/components/tabs/Tab.test.js (hunk 9 above)
Comment (rhenshaw56) [reply tests/components/tabs/Tab.test.js]: resolved!

File: app/js/components/drugOrderEntry/index.jsx (hunk 8 above)
Comment (rhenshaw56) [reply app/js/components/drugOrderEntry/index.jsx]: done!

File: app/js/reducers/orderSelectionReducer.js (hunk 10 above)
Comment (rhenshaw56) [reply app/js/reducers/orderSelectionReducer.js]: done!
